[DS-3977] Dependency updates

mwoodiupui opened the pull request:
https://jira.duraspace.org/browse/DS-3977
Some more vulnerable dependencies, updated.  Passes test suites.

mwoodiupui committed:
[DS-3795] Address vulnerability in Xalan

mwoodiupui committed:
[DS-3795] Remove unnecessary dependency on vulnerable guava artifact.

mwoodiupui committed:
[DS-3795] Update for vulnerable httpclient

mwoodiupui committed:
[DS-3795] Update for vulnerable BouncyCastle artifacts (renamed).

tdonohue commented on pom.xml:
```diff
@@ -1300,13 +1304,12 @@
                 <groupId>xalan</groupId>
                 <artifactId>xalan</artifactId>
-                <version>2.7.0</version>
+                <version>2.7.2</version>
```
FYI, we purposefully downgraded `xalan` to 2.7.0 recently cause of this bug: https://jira.duraspace.org/browse/DS-3733

We probably should add in a comment on this dependency to note that upgrading xalan causes issues with OAI

tdonohue reviewed (commented):
Looks good overall, though I added a comment about the fact that we shouldn't re-upgrade `xalan`.  See above ⬆️

tdonohue commented:
@mwoodiupui : in case you want to add a completely unused dependency to this list (i.e. remove it), I'm almost 100% positive `jargon` (from 2007 it seems) is unused these days:
https://github.com/DSpace/DSpace/blob/master/dspace-api/pom.xml#L331

Figured it might be a good addition to a dependency cleanup PR

mwoodiupui committed:
[DS-3795] Revert upgrade of Xalan due to OAI issues; remove unused decade-old Jargon dependency.

mwoodiupui committed:
[DS-3795] Update zookeeper, used by Solr.

This simplifies the management of the zookeeper version, but also
moves it to dspace-parent where there is no context to explain it.

tdonohue reviewed (approved):
👍 Untested, but changes look good at a glance.

tdonohue commented:
@mwoodiupui : I've removed the reference to ticket DS-3795 from this PR, as that ticket was closed during the 6.3 release. So, this likely needs a new ticket to describe the changes proposed here.  

Also, this PR may need some minor updates (via a rebase) as there's overlaps here with what we just merged in #2158

mwoodiupui committed:
[DS-3795] This was already fixed by another PR.

mwoodiupui committed:
[DS-3795] Another BouncyCastle update for another vulnerability.

tdonohue commented:
Closing, as this is now obsolete & outdated. All these dependencies appear to already be upgraded
